### ISSUE TITLE
Can't read number field where value is greater than 0

### ISSUE THREAD

**Issue by fsandreau** [EddyVerbruggen/nativescript-plugin-firebase#186]
For some reasons, when I do a query request to get my objects, some properties of the json objects are missed. Below the query :

```return firebase.query(
        data => {
            if(data.value) {
                Object.keys(data.value).forEach(key => {
                    this.results.push(data.value[key]); // Here I dont get exactly the JSON, some properties disapears.....
                }
            }
        }, "/ads/", 
        { orderBy: { type: firebase.QueryOrderByType.KEY }})
        .then(result => { return result; })
    );
```

Properties of type number are not loaded (totalProduct and state) when value is greater than 0. <img width="521" alt="capture d ecran 2016-10-19 a 19 02 40" src="https://cloud.githubusercontent.com/assets/793960/19526620/015ae0c8-9636-11e6-8d86-25d093f000a2.png">

Regards

**Comment by jlooper** [comment]
I am experiencing this issue as well. This is really bizarre. I can retrieve strings, not numbers. Should I convert everything to a string? This is crazy. In the example below, everything does load the first time, but then on subsequent loads of the app or when I go to a detail page, anything that is a number is omitted entirely. For example, I need all this data back:

<img width="413" alt="screenshot 2016-12-05 21 57 11" src="https://cloud.githubusercontent.com/assets/1450004/20911454/0d29a93c-bb36-11e6-8c0c-6ef8330f7793.png">

but what I get is only that data that is a string.

**Comment by EddyVerbruggen** [comment]
iOS/Android/both?

I've tested on iOS 10 and Android 7 devices with the demo app as well as the query of @fsandreau and everything works fine with my data structure:

<img width="327" alt="screen shot 2016-12-06 at 11 10 14" src="https://cloud.githubusercontent.com/assets/1426370/20921485/920f7938-bba4-11e6-94a0-e33d5b03c078.png">

Whenever I change fi. the `age` property the data is reflected correctly in my app. Any idea how this is different from your usages?

**Comment by jlooper** [comment]
hi, the very consistently bizarre behavior is that, on initial creation of the app I get the expected data:

[{"id":"-KWdk_ax11wUci2ffQra","PracticesCompleted":5,"Reward":"A special prize!","UID":"ade4AX13CLOlBgW1u1hckPZ7Da02","Date":-1479243094507,"Instrument":4,"AdminPassword":"","TeacherEmail":"user@nativescript.org","Name":"Ellen","PracticeLength":5,"PracticesRequired":6}]

and then, if I livesync or just close and reopen the app on device, I get only the strings:

[{"id":"-KWdk_ax11wUci2ffQra","Reward":"A special prize!","UID":"ade4AX13CLOlBgW1u1hckPZ7Da02","AdminPassword":"","TeacherEmail":"user@nativescript.org","Name":"Ellen"}]

**Comment by EddyVerbruggen** [comment]
@jlooper can I clone some repo to debug this? Also, which platform(s)?

**Comment by jlooper** [comment]
Sorry, it's iOS. I have my repo here: https://github.com/jlooper/practicebuddy-web-mobile

**Comment by EddyVerbruggen** [comment]
Cool, can you point me at the problematic code as well?

And does Android behave correctly?

**Comment by jlooper** [comment]
thanks for your help. The query to get student data is here: https://github.com/jlooper/practicebuddy-web-mobile/blob/master/src/client/app/frameworks/practicebuddy/services/firebase.service.ts#L152-L195

Genymotion is throwing errors all over so I'm battling Android right now. For the moment, though, maybe the above code makes some sense?

**Comment by EddyVerbruggen** [comment]
Ah, so you're not using `query` (@fsandreau does), but `addValueEventListener`. I'll try to look at it later today.

**Comment by jlooper** [comment]
I've created a sample using both query and adding an event listener. Consistently, when I run the app using `tns run ios` I get my data back, but when I use `tns livesync ios --watch` nothing is returned. I would just say, ok, don't use livesync, but I am not convinced that this same problem doesn't occur when I build to device (`tns deploy ios --device device-id`), close the app, and then reopen it. Here's my repo: https://github.com/jlooper/test-tapper

**Comment by EddyVerbruggen** [comment]
@fsandreau Does your issue also only pop up with livesync (on iOS)?

**Comment by EddyVerbruggen** [comment]
@jlooper Seeing similar things. Before I jump to conclusions: can you be very clear (cold hard facts) to me about livesync vs a 'normal' run? Does the problem only pop up when using livesync?

**Comment by EddyVerbruggen** [comment]
@jlooper changing `persist` to `false` in `main.ts` of your test app seems to fix it. So there may be an incompatibility between Firebase persist-mode and livesync.

**Comment by jlooper** [comment]
holy cow. That seems to have done it! Wow, what a strange behavior!! But thank you @eddy and @fsandreau I hope this might help you!!

**Comment by fsandreau** [comment]
Hi all,
Thanks for your work Eddy.
I'm not able to reproduce the same bug without persist mode activated. Its
really wird.
And last think, sorry dor my english, I'm french, but I try to do my best :D

Regards,
Franck

2016-12-06 23:41 GMT+01:00 Jen Looper <notifications@github.com>:

**Comment by andreacappadona17** [comment]
Hi, I'm having the same problem when running my app on iOS (iPad 9,7 2017), both in livesync and deploy mode, with **persist** set to **true**.

Has anyone managed to make their apps run without errors on iOS while keeping persist to true?
Really need to make apps work offline too.

**Comment by EddyVerbruggen** [comment]
Fixed in 4.0.0 thanks to issue #365 (and in retrospect the code posted by @andreacappadona17 which I just saw now).

@jlooper This may render the `persist` workaround obsolete!
